### PR TITLE
Add a test for tab focus handling

### PR DESCRIPTION
**Pull request by hunger**

### CODE REVIEW

**Review by tronical — APPROVED** [review]
Nice idea :)

File: tests/cases/focus/keyboard_focus.slint [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                    return accept;
+                }
+                return reject;
+            }
+            value: 23;
+        }
+
+        for person[i] in model: HorizontalBox {
+            SpinBox {
+                key-pressed(event) => {
```
Comment (ogoffart) [inline tests/cases/focus/keyboard_focus.slint:41]: Aw.....   We should never have exposed that in the public API.

I'd rather you use a FocusScope directly.

Comment (tronical) [reply tests/cases/focus/keyboard_focus.slint:41]: And the second downside of `SpinBox` being a `FocusScope` directly is that when "re-implementing" key-pressed, there's probably no way of calling the original implementation.

We've had this a few times, when a widget in the fluent style wasn't a Rectangle but something else that offered functionality of some kind, like the `GroupBox` being a vertical layout, or `SpinBox` or `ComboBox` being a `FocusScope`.

Maybe it's worth a comment here at least, and when we change `SpinBox` the compiler will hopefully point us to this code needing changing.

**Comment by hunger** [comment]
Also added backward tabbing to all languages.

**Pull request merged**